mrmap/quality/models.py

```diff
@@ -115,40 +108,22 @@ class ConformityCheckConfigurationInternal(ConformityCheckConfiguration):
                                                 blank=True)
 
 
-class ConformityCheckRunManager(models.Manager):
-    """ Custom manager to extend ConformityCheckRun methods """
-
-    def has_running_check(self, metadata: Metadata):
-        """ Checks if the given metadata object has a non-finished
-        ConformityCheckRun.
-
-            Returns:
-                True, if a non-finished ConformityCheckRun was found,
-                false otherwise.
-        """
-        running_checks = super().get_queryset().filter(
-            metadata=metadata, passed__isnull=True).count()
-        return running_checks != 0
-
-    def get_latest_check(self, metadata: Metadata):
-        check = super().get_queryset().filter(metadata=metadata).latest(
-            'time_start')
-        return check
-
-
-class ConformityCheckRun(UuidPk):
+class ConformityCheckRun(CommonInfo, GenericModelMixin):
     """
     Model holding the relation of a metadata record to the results of a check.
     """
-    metadata = models.ForeignKey(Metadata, on_delete=models.CASCADE)
-    conformity_check_configuration = models.ForeignKey(
-        ConformityCheckConfiguration, on_delete=models.CASCADE)
-    time_start = models.DateTimeField(auto_now_add=True)
-    time_stop = models.DateTimeField(blank=True, null=True)
+    metadata = models.ForeignKey(DatasetMetadata, on_delete=models.CASCADE)
+    config = models.ForeignKey(ConformityCheckConfiguration, on_delete=models.CASCADE)
     passed = models.BooleanField(blank=True, null=True)
     result = models.TextField(blank=True, null=True)
 
     objects = ConformityCheckRunManager()
 
+    def get_absolute_url(self):
+        return f'{reverse("quality:conformity_check_run_list")}?id={self.pk}'
+
+    def get_result_url(self):
```

Comment:
could be a part of GenericModelMixin? We need to think about a good way to reuse code for model which are representing  a "celery task" and store there results in an other model. We need this in quality app, monitoring app, csw app... and so on for example.

Reply:
Good idea. Can we have a look at this together?